consensus/hotstuff/validator/validator.go

```diff
@@ -12,74 +12,140 @@ import (
 
 // Validator is responsible for validating QC, Block and Vote
 type Validator struct {
-	// TODO: change to hotstuff.Replicas when front-loading QC verification
-	// https://github.com/onflow/flow-go/pull/2328#discussion_r866494368
-	committee hotstuff.DynamicCommittee
-	forks     hotstuff.ForksReader
+	committee hotstuff.Replicas
 	verifier  hotstuff.Verifier
 }
 
 var _ hotstuff.Validator = (*Validator)(nil)
 
 // New creates a new Validator instance
 func New(
-	committee hotstuff.DynamicCommittee,
-	forks hotstuff.ForksReader,
+	committee hotstuff.Replicas,
 	verifier hotstuff.Verifier,
 ) *Validator {
 	return &Validator{
 		committee: committee,
-		forks:     forks,
 		verifier:  verifier,
 	}
 }
 
-// ValidateQC checks the validity of a QC for a given block. Inputs:
-//  * qc - the qc to be validated
-//  * block - the block that the qc is pointing to
+// ValidateTC validates the TimeoutCertificate `tc`.
 // During normal operations, the following error returns are expected:
-//  * model.InvalidBlockError if the QC is invalid
-func (v *Validator) ValidateQC(qc *flow.QuorumCertificate, block *model.Block) error {
-	if qc.BlockID != block.BlockID {
-		// Sanity check! Failing indicates a bug in the higher-level logic
-		return fmt.Errorf("qc.BlockID %s doesn't match block's ID %s", qc.BlockID, block.BlockID)
+//  * model.InvalidTCError if the TC is invalid
+//  * model.ErrViewForUnknownEpoch if the TC refers unknown epoch
+// Any other error should be treated as exception
+func (v *Validator) ValidateTC(tc *flow.TimeoutCertificate) error {
+	highestQC := tc.NewestQC
```

Comment:
```suggestion
	newestQC := tc.NewestQC
```
suggestion to keep consistent terminology